compiler/modules/front/tests/testParser.bal

```diff
@@ -66,17 +123,33 @@ function reduceToWords(string rule, string fragment) returns err:Syntax|Word[] {
 }
 
 function sourceFragments() returns string[][]|error {
+     string[][] s = check invalidTokenSourceFragments();
+     string[][] valid = check validTokenSourceFragments();
+     foreach var v in valid {
+         s.push(["V", v[2]]);
+     }
+     return s;
+}
+
+function invalidTokenSourceFragments() returns string[][]|error {
+    return [["OE", "\""],
+            ["OE", "'"],
+            ["OE", "`"],
+            ["OE", "\"\\\""],
+            ["OE", "\\"],
+            ["OE", "\"\n\""],
+            ["E", "01"]];
+}
 
+function validTokenSourceFragments() returns string[][]|error {
     string[][] s = 
         [["E", "expr", "", ""],
         // literals
          ["V", "expr", "0", "0"],
          ["V", "expr", "1", "1"],
-         ["E", "expr", "01", ""],
          ["V", "expr", "()", "()"],
          ["V", "expr", "-()", "-()"],
          ["V", "expr", "-true", "-true"],
-         ["E", "expr", "\"", ""],
```

Comment:
I suggest you use a string template when the source includes Ballerina strings: otherwise quoting will get confusing.